plugins/schedule/base/lib/kScheduleEventsConsumer.php

```diff
@@ -120,19 +172,60 @@ public function updateScheduleEvent($eventId)
         }
     }
 
-    protected function scheduleEventChanged(ScheduleEvent $scheduleEvent, $modifiedColumns)
-    {
-        if (in_array($scheduleEvent->getStatus(), array(ScheduleEventStatus::DELETED, ScheduleEventStatus::CANCELLED)))
-        {
-            $scheduleEvents = ScheduleEventResourcePeer::retrieveByEventId($scheduleEvent->getId());
-            foreach ($scheduleEvents as $currScheduleEvent)
-            {
-                /**
-                 * @var ScheduleEventResource $currScheduleEvent
-                 */
-                $currScheduleEvent->delete();
-            }
-        }
-    }
+	protected function scheduleEventChanged(ScheduleEvent $scheduleEvent, $modifiedColumns = null)
+	{
+		if (in_array($scheduleEvent->getStatus(), array(ScheduleEventStatus::DELETED, ScheduleEventStatus::CANCELLED)))
+		{
+			$scheduleEvents = ScheduleEventResourcePeer::retrieveByEventId($scheduleEvent->getId());
+			foreach ($scheduleEvents as $currScheduleEvent)
+			{
+				/**
+				 * @var ScheduleEventResource $currScheduleEvent
+				 */
+				$currScheduleEvent->delete();
+			}
+			if ($scheduleEvent->getLinkedTo() && !is_null($scheduleEvent->getLinkedTo()->getEventId()))
+			{
+				$scheduleEvent->removeCurrentEventFromPrecedingEvent();
+			}
+			if ($scheduleEvent->getLinkedBy())
+			{
+				$scheduleEvent->unlinkFollowerEvents();
+			}
+		}
+		if ($modifiedColumns)
+		{
+			if (in_array(ScheduleEventPeer::END_DATE, $modifiedColumns))
+			{
+				//update start & end date for all linked by events
+				$scheduleEvent->updateStartEndTimeOfFollowerEvents();
+				if ($scheduleEvent->getLinkedTo() && $scheduleEvent->getLinkedTo()->getEventId())
+				{
+					$linkedToStartDate = $scheduleEvent->getLinkedToEndTime();
+					$offset = $scheduleEvent->getLinkedTo()->getOffset();
+					if (strtotime($scheduleEvent->getStartDate()) != strtotime($linkedToStartDate) + $offset)
+					{
+						$scheduleEvent->removeCurrentEventFromPrecedingEvent();
+						$scheduleEvent->setLinkedTo(null);
+						$scheduleEvent->save();
+					}
+				}
+			}
+			if (in_array(ScheduleEventPeer::CUSTOM_DATA, $modifiedColumns) && in_array(ScheduleEventPeer::UPDATED_AT, $modifiedColumns))

```

Comment:
ScheduleEventPeer::UPDATED_AT is not relevant here. can. use function isCustomDataModified to check if the linkedTo has changed

Reply:
done